packages/esm-patient-medications-app/src/components/medications-details-table.component.tsx

```diff
@@ -84,13 +84,13 @@ const MedicationsDetailsTable = connect<
           <div className={styles.medicationRecord}>
             <div>
               <p className={styles.bodyLong01}>
-                <strong>{capitalize(medication.drug?.name)}</strong> &mdash; {medication.drug?.strength.toLowerCase()}{' '}
-                &mdash; {medication.doseUnits?.display.toLowerCase()}
+                <strong>{capitalize(medication.drug?.concept?.display)}</strong> &mdash;{' '}
+                {medication.drug?.strength.toLowerCase()} &mdash; {medication.drug.dosageForm.display.toLowerCase()}
               </p>
               <p className={styles.bodyLong01}>
                 <span className={styles.label01}>{t('dose', 'Dose').toUpperCase()}</span>{' '}
                 <span className={styles.dosage}>
-                  {getDosage(medication.drug?.strength, medication.dose).toLowerCase()}
+                  {medication.dose} {medication.doseUnits.display.toLowerCase()}
```

Comment:
@samuelmale this is where the logic is replaced - the problematic logic was introduced by the getDosage function, which is the one causing the multiplier. Do you think we need an additional/deeper fix as well?

Reply:
Oohhh I see thanks Grace. (It wouldn't be sarcastic if someone requested you to review a PR with chunks of commits  think you should start writing some code)

@vasharma05 so what's gonna happen to malicious `getDosage(...)` function? Is left as dead code? Is it being consumed in a different place?

Reply:
Removed the dead import. Thanks @samuelmale !